gprofiler/profilers/java.py

```diff
@@ -768,6 +877,19 @@ def _init_java_safemode(self, java_safemode: str) -> None:
                 f" --java-safemode={JavaSafemodeOptions.JAVA_EXTENDED_VERSION_CHECKS}"
             )
 
+    def _init_java_flag_collection(self, java_jvm_flags_to_collect: str) -> Union[JavaFlagCollectionOptions, List[str]]:
+        if java_jvm_flags_to_collect is None or java_jvm_flags_to_collect == "":
+            return JavaFlagCollectionOptions.DEFAULT
+        if any(
+            [
+                java_jvm_flags_to_collect == java_flag_collection_option.value
+                for java_flag_collection_option in JavaFlagCollectionOptions
+            ]
+        ):
+            return JavaFlagCollectionOptions(java_jvm_flags_to_collect)
+        else:
+            return java_jvm_flags_to_collect.split(",")
```

Comment:
Is there nothing built-in argparse to do that for you?

Reply:
thats how we do it for safemode options.[
](https://github.com/Granulate/gprofiler/blob/b07653ccba6b3536a6d94c222204e74f7b1e1a5f/gprofiler/profilers/java.py#L865)